### PR TITLE
Get rid of lints with performance implications

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Some alternatives are more efficient or avoid unnecessary function calls.

### CODE REVIEW

File: R/detect-alignment.R [hunk 1]
```diff
@@ -104,10 +104,10 @@ token_is_on_aligned_line <- function(pd_flat) {
   # now, pd only contains arguments separated by values, ideal for iterating
   # over columns.
   n_cols <- map_int(pd_by_line, ~ sum(.x$token == "','"))
-  previous_line <- 0
-  current_col <- 0
-  start_eval <- ifelse(alignment_col1_all_named(pd_by_line), 1, 2)
-  for (column in seq2(1, max(n_cols))) {
+  previous_line <- 0L
+  current_col <- 0L
```
Comment (lorenzwalthert) [inline R/detect-alignment.R:108]: I think we could even `x <- y <- 1` no?

**Review by lorenzwalthert — APPROVED** [review]
looks good overall, not so much speed improvement unfortunately, but cleaner code 👍 .

File: R/detect-alignment.R (hunk 1 above)
Comment (IndrajeetPatil) [reply R/detect-alignment.R]: Done, thanks.

**Comment by IndrajeetPatil** [comment]
> not so much speed improvement unfortunately

With time, they should accumulate to make a difference 🤞

**Comment by lorenzwalthert** [comment]
> With time, they should accumulate to make a difference 🤞

Yes, for cummulated effects, there is https://github.com/r-lib/styler/pull/868 to compare to last release. Already a few percentage points faster 👍

**Pull request merged**